app/models/merchant.rb

```diff
@@ -6,4 +6,9 @@ class Merchant < ApplicationRecord
   def invoices
     Invoice.joins(items: :merchant).where(items: { merchant: self }).distinct
   end
+
+  def favorite_customers
+    ids = invoices.pluck(:id)   
+    Customer.select("first_name, last_name, count(result)").joins(invoices: :transactions).where(transactions: {result: "success"}).where(invoices: {id: ids}).order(count: :desc).group("first_name, last_name").limit(5)
+  end
```

Comment:
This looks good! 
If you find it helpful in the future, you can start a new line before each command for better visibility.

E.g. 
```
Customer.select("first_name, last_name, count(result)")
  .joins(invoices: :transactions)
  .where(transactions: {result: "success"})
  .where(invoices: {id: ids})
  .order(count: :desc).group("first_name, last_name")
  .limit(5)
```

Reply:
Wow thanks @westonio , I didn't know that! I do like the visibility here